<!-- <?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="example">

    <link name="link_0">
        <visual>
            <geometry>
                <mesh filename="package://example_description/meshes/link_0.stl"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
        </visual>
    </link>

    <link name="link_1">
        <visual>
            <geometry>
                <mesh filename="package://example_description/meshes/link_1.stl"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
        </visual>
    </link>

    <joint name="link_0_to_link_1" type="revolute">
        <axis xyz="0 0 1"/>
        <limit 
            effort="1000.0" 
            lower="-3.14" 
            upper="3.14" 
            velocity="0.5"/>

        <parent link="link_0"/>
        <child link="link_1"/>
        <origin xyz="0 0 0.2"/>
    </joint>

    <link name="link_2">
        <visual>
            <geometry>
                <mesh filename="package://example_description/meshes/link_2.stl"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
        </visual>
    </link>

    <joint name="link_1_to_link_2" type="revolute">
        <axis xyz="0 0 1"/>
        <limit 
            effort="1000.0" 
            lower="-0.548" 
            upper="0.548" 
            velocity="0.5"/>
            
        <parent link="link_1"/>
        <child link="link_2"/>
        <origin rpy="0 -1.57 -1.57" xyz="0 -0.12 0"/>
    </joint>

    <link name="link_3">
        <visual>
            <geometry>
                <mesh filename="package://example_description/meshes/link_3.stl"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0 0 0"/>
        </visual>
    </link>

    <joint name="link_2_to_link_3" type="revolute">
        <axis xyz="0 0 1"/>
        <limit 
            effort="1000.0" 
            lower="-0.548" 
            upper="0.548" 
            velocity="0.5"/>
            
        <parent link="link_2"/>
        <child link="link_3"/>
        <origin rpy="0 0 0" xyz="0.25 0 0.1"/>
    </joint>

    <link name="end_effector">
        <visual>
            <geometry>
                <mesh filename="package://example_description/meshes/end_effector.stl"/>
            </geometry>
            <origin rpy="0 0 1.57" xyz="0 0 0"/>
        </visual>
    </link>

    <joint name="link_3_to_end_effector" type="revolute">
        <axis xyz="0 0 1"/>
        <limit 
            effort="1000.0" 
            lower="-0.548" 
            upper="0.548" 
            velocity="0.5"/>
            
        <parent link="link_3"/>
        <child link="end_effector"/>
        <origin rpy="0 1.57 0" xyz="0.28 0 0"/>
    </joint>

    

</robot> -->


